sql/2022/jamstack/distribution_age_headers_desktop.sql

```diff
@@ -0,0 +1,33 @@
+-- getting ages to calculate median age (or any other percentage threshold)
+-- we're using 2022-06-01 run as the baseline because values change every year
```

Comment:
This looks at the age of ALL requests. Including analytics pixesl and the like which may be short.

We should restrict this to home page only surely? Using the `almanac dataset` makes this easier as it has a `firstHTML` field and also pulls the `age` field out for querying.

We also usually include both desktop and mobile in the one query:

```sql
SELECT
  client,
  date,
  COUNT(0) AS total,
  COUNTIF(resp_age IS NOT NULL AND resp_age != '') AS age,
  COUNTIF(resp_age IS NULL OR resp_age = '') AS no_age,
  COUNTIF(resp_age IS NOT NULL AND resp_age != '') / COUNT(0) AS pct_age,
  COUNTIF(resp_age IS NULL OR resp_age = '') / COUNT(0) AS pct_no_age,
FROM
  `httparchive.almanac.requests`
WHERE
  firstHtml
GROUP BY
  client,
  date
ORDER BY
  client,
  date
```